pycbc/events/eventmgr.py

```diff
@@ -251,14 +251,22 @@ def keep_near_injection(self, window, injections):
         i = indices_within_times(gpstime, inj_time - window, inj_time + window)
         self.events = self.events[i]
 
-    def keep_loudest_in_interval(self, window, num_keep, log_chirp_width=None):
+    def keep_loudest_in_interval(self, window, num_keep, statname="newsnr",
+                                 log_chirp_width=None):
         if len(self.events) == 0:
             return
 
-        e = self.events
-        # FIXME allow statistics that are not newsnr
-        stat = ranking.newsnr(abs(e['snr']), e['chisq'] / e['chisq_dof'])
-        time = e['time_index']
+        from pycbc.events import stat
+        e_copy = self.events
+
+        # Messy step because pycbc inspiral's internal 'chisq_dof' is 2p-2
+        # but stat.py / ranking.py functions use 'chisq_dof' = p
+        e_copy['chisq_dof'] = e_copy['chisq_dof'] / 2 + 1
```

Comment:
I don't think this will actually copy `self.events` ... So I think this step will also change `self.events['chisq_dof']` .... I might suggest to just change `chisq_dof`, perform this step, and then set it back after doing `self.events = self.events[keep]`